meter.go

```diff
@@ -224,6 +246,12 @@ func (ma *meterArbiter) tick() {
 	}
 }
 
+// should only be called with Lock() held
+func (ma *meterArbiter) newID() int64 {
+	ma.id++
```

Comment:
That also made me a bit concerned.
The only reason why we want a map is to have constant-time deletion, so we don't actually need the concept of "IDs" for metrics.
Couldn't we register the meter in the arbiter's map as the pointer to itself and avoid ID generation? We could also add a check to `stopped` to make sure it gets done only once.